src/main/java/org/cloudfoundry/reconfiguration/tomee/GenericServicePropertiesProvider.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2016-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.cloudfoundry.reconfiguration.tomee;
+
+import org.apache.openejb.api.resource.PropertiesResourceProvider;
+import org.springframework.cloud.cloudfoundry.CloudFoundryRawServiceData;
+import org.springframework.cloud.cloudfoundry.com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.cloud.util.EnvironmentAccessor;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * Created by tomcollings on 1/21/18.
+ */
+public class GenericServicePropertiesProvider implements PropertiesResourceProvider {
+
+    private ObjectMapper objectMapper;
+    private EnvironmentAccessor environment;
```

Comment:
For `environment` may be you can make it final and provide a constructor with package visibility then in the test you will use the constructor instead of reflection?

Reply:
Correct.  Two constructors:

```java
public GenericServicePropertiesProvider() {
    this(new EnvironmentAccessor(), new ObjectMapper());
}

GenericServicePropertiesProvider(Environment environment, ObjectMapper objectMapper) {
    this.environment = environment;
    this.objectMapper = objectMapper;
}
```